sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2120,6 +2122,99 @@ class Analyzer(
     }
   }
 
+  /**
+   * Resolves columns of an output table from the data in a logical plan. This rule will:
+   *
+   * - Reorder columns when the write is by name
+   * - Insert safe casts when data types do not match
+   * - Insert aliases when column names do not match
+   * - Detect plans that are not compatible with the output table and throw AnalysisException
+   */
+  object ResolveOutputRelation extends Rule[LogicalPlan] {
+    override def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case append @ AppendData(table: NamedRelation, query, isByName)
```

Comment:
Since we don't have an analyzer rule for `ApendData(LogicalPlan ...)`, shall we just require `AppendData.table` to be a `NamedRelation` in class definition?

Reply:
Yes, I agree.